roles/reportportal/tasks/main.yaml

```
@@ -0,0 +1,79 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if mandatory variables are defined and not empty
+  ansible.builtin.assert:
+    that:
+      - cifmw_reportportal_datarouter_url is defined and cifmw_reportportal_datarouter_url != ""
+      - cifmw_reportportal_datarouter_username is defined and cifmw_reportportal_datarouter_username != ""
+      - cifmw_reportportal_datarouter_password is defined and cifmw_reportportal_datarouter_password != ""
+      - cifmw_reportportal_project is defined and cifmw_reportportal_project != ""
+      - cifmw_reportportal_droute_client_url is defined and cifmw_reportportal_droute_client_url != ""
```

It's considered a best practice (especially for code clarity) to avoid comparing a string to "empty" and rather check for its length.

```suggestion
      - cifmw_reportportal_datarouter_url is defined and cifmw_reportportal_datarouter_url | length > 0
      - cifmw_reportportal_datarouter_username is defined and cifmw_reportportal_datarouter_username | length > 0
      - cifmw_reportportal_datarouter_password is defined and cifmw_reportportal_datarouter_password | length > 0
      - cifmw_reportportal_project is defined and cifmw_reportportal_project | length > 0
      - cifmw_reportportal_droute_client_url is defined and cifmw_reportportal_droute_client_url | length > 0
```

---

Done

---

In fact, there's a linter rule for that, that I assume is not enabled if it's not blaming

---

correct, it's an opt-in rule